### PR TITLE
Strong version of dmap_uni

### PR DESCRIPTION
**Pull request by fdupress**

### CODE REVIEW

File: theories/distributions/Distr.ec [hunk 1]
```diff
@@ -908,9 +908,22 @@ lemma dmap_ll (d : 'a distr) (f : 'a -> 'b) :
...
+proof.
+  move=> finj_in duni x y /supp_dmap [a [ina ->]] /supp_dmap [b [inb ->]].
+  rewrite !dmap1E.
+  have Heq: forall z, z \in d => mu d (pred1 (f z) \o f) = mu d (pred1 z).
+  + move=> z inz; rewrite (@mu_eq_support d (pred1 (f z) \o f) (pred1 z)) //.
+    move=> x0 inx0 @/(\o) @/pred1 /=; rewrite eq_iff=> />; exact/finj_in.
+  by rewrite !Heq //; exact/duni.
+qed.
+
 lemma dmap_uni (d : 'a distr) (f : 'a -> 'b) :
```
Comment (strub) [inline theories/distributions/Distr.ec]: Do reprove `dmap_uni` as a direct corollary of `dmap_uni_support`.

**Comment by strub** [comment]
Is this PR the same as #30 ?

**Comment by fdupress** [comment]
Yes. I'll close when I'm done lecturing.

François

**Comment by strub** [comment]
Ok, I merged your proof that was more elegant in #30 and I am going to merge #30.

**Pull request closed without merging**